### PR TITLE
Raise an exception when an L2cap connection fails

### PR DESCRIPTION
**Pull request by benquike**

### CODE REVIEW

File: bumble/l2cap.py [hunk 1]
```diff
@@ -2078,7 +2078,8 @@ async def connect(self, connection: Connection, psm: int) -> Channel:
         # Connect
         try:
             await channel.connect()
-        except Exception:
+        except Exception as e:
```
Comment (barbibulle) [inline bumble/l2cap.py:2081]: maybe use  `try: xxx finally: del yyy` ?

**Review by barbibulle — APPROVED** [review]

File: bumble/l2cap.py (hunk 1 above)
Comment (benquike) [reply bumble/l2cap.py:2081]: moving del to finally would make it wrong, right?
we only want to delete when exception happens?

Comment (uael) [reply bumble/l2cap.py:2081]: @benquike No you could move the delete to finally 👌

Comment (uael) [reply bumble/l2cap.py:2081]: I miss-read it!

**Pull request merged**